src/adapter/src/catalog/open.rs

```diff
@@ -382,30 +382,6 @@ impl Catalog {
                 schemas_by_name.insert(name.clone(), id);
             }
 
-            let roles = txn.get_roles();
-            for mz_catalog::durable::Role {
-                id,
-                name,
-                attributes,
-                membership,
-                vars,
-            } in roles
-            {
-                let oid = state.allocate_oid()?;
-                state.roles_by_name.insert(name.clone(), id);
-                state.roles_by_id.insert(
-                    id,
-                    Role {
-                        name,
-                        id,
-                        oid,
-                        attributes,
-                        membership,
-                        vars,
-                    },
-                );
-            }
-
             let default_privileges = txn.get_default_privileges();
             for mz_catalog::durable::DefaultPrivilege { object, acl_item } in default_privileges {
                 state.default_privileges.grant(object, acl_item);
```

Comment:
You'll have to move default and system privileges down too. These both rely on the roles already existing in the catalog and will panic or fail if the roles don't exist yet.

Reply:
As discussed, this isn't true. `grant` doesn't actually attempt to look up the role IDs.